test/optimizer/table_or_pushdown.test

```diff
@@ -0,0 +1,96 @@
+# name: test/optimizer/table_or_pushdown.test
+# description: Test Table OR Filter Push Down
+# group: [optimizer]
+
+statement ok
```

Comment:
Could we add some tests with other scanners that perform filter pushdown, e.g.:

* The Parquet scanner
* The Arrow scanner (as Pedro mentioned)